src/lua/socket.lua

```diff
@@ -952,6 +952,22 @@ local function socket_new(domain, stype, proto)
     end
 end
 
+local function socket_from_fd(fd)
```

Comment:
Shouldn't we check that the socket is non-blocking? Otherwise the socket object methods (such as `:read()`) may work in an unexpected way.

And since we're owning the file descriptor now, I guess it is legal to set the non-blocking mode here.

Reply:
The function doesn't perform any checks on the given fd. One may pass an invalid fd or fd that represents a file, in which case the `read` method may fail or block. It is by [design](https://www.notion.so/tarantool/IPROTO-over-socketpair-fb5a30dc42a348689655bf75248d9020?pvs=4#f5320789c81e466eb2632073ed47bb7c).

As for switching the fd to the non-blocking, it would be redundant in most cases because the fd returned by `socket:fd()` should already be in the non-blocking mode, no matter if the socket was created with `socket()` or `socket.socketpair()`. Note that the non-blocking mode is inherited by the child process after fork.

Reply:
OK, just highlighted this point, because `socket()` performs this action. Maybe worth to highlight it in the documentation.

Anyway, I'm OK with the answer and closing the discussion.